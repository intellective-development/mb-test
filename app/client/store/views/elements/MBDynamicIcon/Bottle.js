// @flow

import * as React from 'react';

const Bottle = () => (
  <path
    d="M50 95V60h20v35zM40 110h40V45L70 35V10H50v25L40 45zm10-40h20M50 20h20"
    fill="none"
    strokeLinejoin="round"
    strokeWidth="4" />
);

export default Bottle;
