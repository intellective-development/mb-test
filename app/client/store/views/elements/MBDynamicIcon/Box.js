// @flow

import * as React from 'react';

const Box = () => (
  <path
    d="M24 96V20l24 8v76zm0-76l48-8 24 8v76l-48 8m0-76l48-8"
    fill="none"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="4" />
);

export default Box;
