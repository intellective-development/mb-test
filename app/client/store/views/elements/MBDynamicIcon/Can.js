// @flow

import * as React from 'react';

const Can = () => (
  <path
    d="M40 10h40c4 1 4 6 0 7H40c-4-1-4-6 0-7zm0 7l-9 15-1 56c0 17 0 15 13 15m37-86l9 15v56c0 19 3 15-38 15m-16 0c1 8 3 7 26 7 21 0 22 1 24-7M31 33h12m8 0h38"
    fill="none"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="4" />
);

export default Can;
