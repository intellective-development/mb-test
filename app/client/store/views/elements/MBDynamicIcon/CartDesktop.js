// @flow

import * as React from 'react';

const CartDesktop = () => (
  <g fill="#222" fillRule="evenodd">
    <path d="M12.55 28.067c1.243 0 2.25 1.025 2.25 2.289 0 1.264-1.007 2.289-2.25 2.289-1.242 0-2.249-1.025-2.249-2.29 0-1.263 1.007-2.288 2.25-2.288M27.677 28.067c1.242 0 2.249 1.025 2.249 2.289 0 1.264-1.007 2.289-2.25 2.289-1.242 0-2.249-1.025-2.249-2.29 0-1.263 1.007-2.288 2.25-2.288" />
    <path fillRule="nonzero" d="M5.323 5.402L9.565 24.01h20.89L35.64 10l-1.917.003-4.517 12.189H10.997L6.756 3.584h-4.11L.674 5.642l1.293 1.264L3.41 5.402z" />
    <path d="M29.102 17.136l-.676 1.819H12.373l-.419-1.819h17.148zM30.919 12.245l-.555 1.492-.124.326H11.247l-.418-1.818h20.09z" fillRule="nonzero" />
  </g>
);

export default CartDesktop;
