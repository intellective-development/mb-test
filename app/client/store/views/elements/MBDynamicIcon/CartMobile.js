// @flow

import * as React from 'react';

const CartMobile = () => (
  <g fill="#222" fillRule="evenodd">
    <path d="M13.5 24.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zM23 23a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3" />
    <path fillRule="nonzero" d="M6.549 6.944L9.664 20h15.864L29.5 9.801l-2.143-.079-3.233 8.334h-12.85L8.16 5H4.051L2.5 6.547l1.455 1.35.955-.953z" />
    <path d="M23.5 14l-.748 2h-9.79l-.462-2h11zm1.511-4l-.748 2H11.985l-.462-2H25.01zm1.5 1.843V12h-.06l.06-.157z" />
  </g>
);

export default CartMobile;
