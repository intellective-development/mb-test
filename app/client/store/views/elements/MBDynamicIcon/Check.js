// @flow

import * as React from 'react';

const Check = () => (
  <g fillRule="nonzero">
    <path d="M13.797 20.044l9.574-9.574 1.061 1.06-10.635 10.636-5.327-5.327 1.06-1.06z" />
  </g>
);

export default Check;
