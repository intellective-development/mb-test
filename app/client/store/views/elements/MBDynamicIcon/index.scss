@import "../../style/variables";

.elMBDynamicIcon {
  display: flex;
}

.elMBDynamicIcon__black {
  fill: $brand-black;
}

.elMBDynamicIcon__white {
  fill: $brand-white;
}

.elMBDynamicIcon__red {
  fill: $brand-red;
}
