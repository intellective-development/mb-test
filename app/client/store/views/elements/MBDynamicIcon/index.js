// @flow

import * as React from 'react';
import bindClassNames from '../../../../shared/utils/bind_classnames';

import Bottle from './Bottle';
import Box from './Box';
import Can from './Can';
import CartDesktop from './CartDesktop';
import CartMobile from './CartMobile';
import Check from './Check';
import ChevronRight from './ChevronRight';
import ChevronLeft from './ChevronLeft';
import Email from './Email';
import FAQ from './FAQ';
import Hamburger from './Hamburger';
import Keg from './Keg';
import Phone from './Phone';
import Representative from './Representative';
import RightArrow from './RightArrow';
import Search from './Search';
import SMS from './SMS';
import XClose from './XClose';
import styles from './index.scss';

const cn = bindClassNames(styles);

const DEFAULT_VIEW_BOX = '0 0 32 32';

// TODO: look into the view box, this doesn't seem ideal.
// Should we perhaps store a base height/width or an aspect ratio for each of these?
const ICONS = {
  bottle: { Icon: Bottle, view_box: '0 0 120 120' },
  bottles: { Icon: Bottle, view_box: '0 0 120 120' },
  box: { Icon: Box, view_box: '0 0 120 120' },
  btl: { Icon: Bottle, view_box: '0 0 120 120' },
  can: { Icon: Can, view_box: '0 0 120 120' },
  cans: { Icon: Can, view_box: '0 0 120 120' },
  cart_desktop: { Icon: CartDesktop, view_box: '0 0 40 34' },
  cart_mobile: { Icon: CartMobile, view_box: DEFAULT_VIEW_BOX },
  check: { Icon: Check, view_box: DEFAULT_VIEW_BOX },
  chevron_right: { Icon: ChevronRight, view_box: DEFAULT_VIEW_BOX },
  chevron_left: { Icon: ChevronLeft, view_box: DEFAULT_VIEW_BOX },
  email: { Icon: Email, view_box: DEFAULT_VIEW_BOX },
  faq: { Icon: FAQ, view_box: DEFAULT_VIEW_BOX },
  hamburger: { Icon: Hamburger, view_box: DEFAULT_VIEW_BOX },
  keg: { Icon: Keg, view_box: '0 0 120 120' },
  phone: { Icon: Phone, view_box: DEFAULT_VIEW_BOX },
  representative: { Icon: Representative, view_box: DEFAULT_VIEW_BOX },
  right_arrow: { Icon: RightArrow, view_box: DEFAULT_VIEW_BOX },
  search: { Icon: Search, view_box: DEFAULT_VIEW_BOX },
  sms: { Icon: SMS, view_box: DEFAULT_VIEW_BOX },
  x_close: { Icon: XClose, view_box: DEFAULT_VIEW_BOX }
};

type IconName = $Keys<typeof ICONS>;

type MBDynamicIconProps = {
  name: IconName,
  className?: string,
  width: number,
  height: number,
  color?: 'white' | 'black' | 'red' | 'transparent'
};

const MBDynamicIcon = ({name, width, height, color, className, ...rest_props}: MBDynamicIconProps) => {
  const classes = cn('elMBDynamicIcon', className);
  const colorClass = cn({ [`elMBDynamicIcon__${String(color)}`]: color });
  const { Icon, view_box } = ICONS[name] || ICONS.bottle;

  return (
    <div {...rest_props} className={classes}>
      <svg width={width} height={height} className={colorClass} viewBox={view_box} xmlns="http://www.w3.org/2000/svg">
        <Icon name={name} />
      </svg>
    </div>
  );
};

export default MBDynamicIcon;
