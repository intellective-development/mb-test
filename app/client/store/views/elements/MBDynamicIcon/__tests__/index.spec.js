import * as React from 'react';
import MBDynamicIcon from '../index';

describe('MBDynamicIcon', () => {
  it('renders', () => {
    expect(shallow(<MBDynamicIcon name="email" width={25} height={25} color="mb_white" />)).toMatchSnapshot();
  });
});
