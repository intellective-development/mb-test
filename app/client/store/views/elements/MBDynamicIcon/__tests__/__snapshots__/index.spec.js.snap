// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBDynamicIcon renders 1`] = `
<div
  className="elMBDynamicIcon"
>
  <svg
    className="elMBDynamicIcon__mb_white"
    height={25}
    viewBox="0 0 32 32"
    width={25}
    xmlns="http://www.w3.org/2000/svg"
  >
    <Email
      name="email"
    />
  </svg>
</div>
`;
